iothub/device/src/Authentication/DeviceAuthenticationWithSakRefresh.cs

```diff
@@ -10,18 +10,18 @@ namespace Microsoft.Azure.Devices.Client
     // Implementing SAS Token refresh based on a SharedAccessKey (SAK).
     internal class DeviceAuthenticationWithSakRefresh : DeviceAuthenticationWithTokenRefresh
     {
-        private readonly IotHubConnectionString _connectionString;
+        private readonly IotHubConnectionInfo _connectionString;
```

Comment:
I think the name was misleading. It isn't a connection string, rather it parses them, holds the info, produces one, and generates new passwords based on the auth information it was provided.

Also, it was split into 2 files, for no apparent reason. I consolidated them.